js/util/script_detection.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+module.exports.allowsIdeographicBreaking = function(input) {
+    for (let i = 0; i < input.length; i++) {
+        if (!exports.charAllowsIdeographicBreaking(input.charCodeAt(i), input.charCodeAt(i + 1))) {
+            return false;
+        }
+    }
+    return true;
+};
+
+
+module.exports.charAllowsIdeographicBreaking = function(char, nextChar) {
+    // "一" to "鿌"
+    if (char >= 0x4E00 && char <= 0x9FCC) return true;
+
+    // "㐀" to "䶵"
+    if (char >= 0x3400 && char <= 0x4DB5) return true;
+
+    // eslint-disable-next-line no-irregular-whitespace
+    // "　" to "〿"
+    if (char >= 0x3000 && char <= 0x303F) return true;
+
+    // "𠀀" to "𬺯"
+    if (char === 0xD840 && nextChar >= 0xDC00) return true;
+    if (char >= 0xD841 && char <= 0xD872) return true;
+    if (char === 0xD873 && nextChar <= 0xDEAF) return true;
```

Comment:
I've realized that there are a few bugs in our handling of surrogate pairs. Because we don't actually have the ability to render surrogate pairs as glyphs, I can't make a test case. I'm thinking we should remove this range altogether.